lib/perl/Genome/Sys/LockProxy.pm

```diff
@@ -0,0 +1,106 @@
+package Genome::Sys::LockProxy;
+
+use strict;
+use warnings;
+
+use Genome::Sys::Lock qw();
+use Params::Validate qw(validate_with);
+use Scalar::Util qw(blessed);
+
+=item new()
+
+Keyword Arguments:
+
+=over
+
+=item resource
+
+C<resource> is a unique identifier for some resource.
+
+=item scope
+
+C<scope> is the scope to which the lock is bound.  See C<Genome::Sys::Lock::scopes()> for valid scopes.
+
+=cut
+
+sub new {
+    my $class = shift;
+    my $spec = Genome::Sys::Lock::LOCK_RESOURCE_PARAMS_SPEC();
+    $spec->{resource} = delete $spec->{resource_lock};
+    my %params = validate_with(
+        params => \@_,
+        spec => $spec,
+    );
+    return bless \%params, $class;
```

Comment:
This will crash if `new()` is called on an existing object.  Should we also validate that `$class` is not a reference?